src/condor_utils/MapFile.cpp

```diff
@@ -712,7 +712,7 @@ bool CanonicalMapPrefixEntry::matches(const char * principal, int /*cch*/, std::
 			if(pcanon) { *pcanon = value; }
 			if(groups) {
 				groups->clear();
-				groups->push_back(value);
+				groups->push_back(key.c_str());
```

Comment:
Um? Yikes.

Reply:
Prefix matches aren't regular expressions, so this is not a semantic change.  The previous code returned something you already knew -- the entire URL.  This code returns something you don't know -- the prefix it matched against.  This allows you to avoid repeating yourself in the map file:

```*    pelican://example.vo/example.fs/    plugin.sh $0```